File: src/Games.cs
namespace PetusLauncher;

record ChangelogEntry(string Version, string Notes);

record GameDef(
    string Id,
    string Name,
    string Type,        // "gdps" | "mc"
    string Tagline,
    string? Ip,
    string Description,     // long store-page description
    bool HashCheck,        // integrity/hash verification available (GDPS only)
    ChangelogEntry[] Changelog
);

static class Games
{
    public static readonly GameDef[] All =
    {
        new GameDef(
            "petusgdps", "PetusGDPS", "gdps",
            "Приватный сервер Geometry Dash 2.2",
            null,
            "PetusGDPS — приватный сервер Geometry Dash 2.2 с собственным ядром, " +
            "рейтингами, уровнями и системой опыта Better Progression. В сборку " +
            "входят 23 мода: GMD/Editor API, QOLMod, загрузчик текстур и другое. " +
            "Вход только через Petus ID — пароли не нужны.",
            true,
            new[]
            {
                new ChangelogEntry("1.0.7", "Better Progression синхронизирует опыт с сервером Петус. Исправлены уровни и музыка."),
                new ChangelogEntry("1.0.6", "23 мода, вкладка Mods в Options, токен-авторизация."),
                new ChangelogEntry("1.0.1", "Приветственные текстбоксы, бонус новичкам."),
                new ChangelogEntry("1.0.0", "Первый релиз PetusGDPS."),
            }
        ),
        new GameDef(
            "petusmc", "PetusMC", "mc",
            "Анархический Minecraft-сервер экосистемы Петус",
            "mc.petus.ru",
            "PetusMC — анархический Minecraft-сервер. Античит практически " +
            "отсутствует, правил нет: читы разрешены, можно как угодно " +
            "модифицировать свой Minecraft — мы не проверяем хэши файлов. " +
            "Версия сервера — 1.21.11. Выбери версию клиента (1.21+), войди " +
            "через аккаунт Microsoft или как офлайн-игрок, и запускай прямо " +
            "из лаунчера.",
            false,
            new[]
            {
                new ChangelogEntry("live", "Сервер онлайн (1.21.11). Заходи по mc.petus.ru."),
            }
        ),
    };
}

File: src/main/games.js
// Catalogue of games shown in the launcher sidebar. Each entry drives its
// store-style page. 'gdps' games are installed/updated + launched by the
// launcher; 'mc' games just show a joinable server IP.
'use strict';

const CORE = process.env.PETUS_CORE_URL || 'https://cgdps.petus.ru';

module.exports = [
  {
    id: 'petusgdps',
    name: 'PetusGDPS',
    type: 'gdps',
    tagline: 'Приватный сервер Geometry Dash 2.2',
    // Manifest served by the core (Kestrel, no CDN cache).
    updateManifest: `${CORE}/api/game/manifest`,
    // Banner/logo are bundled assets (fall back to a gradient if missing).
    banner: 'assets/banner-gdps.png',
    icon: 'assets/icon.png',
    changelog: [
      { version: '1.0.3', notes: '21 модов: GMD/Editor API, QoL, текстуры. Вкладка Mods в Options.' },
      { version: '1.0.1', notes: 'Приветственные текстбоксы, бонус новичкам.' },
      { version: '1.0.0', notes: 'Первый релиз PetusGDPS.' },
    ],
  },
  {
    id: 'petusmc',
    name: 'PetusMC',
    type: 'mc',
    tagline: 'Minecraft-сервер экосистемы Петус',
    ip: 'mc.petus.ru',
    banner: 'assets/banner-mc.png',
    icon: 'assets/icon.png',
    changelog: [
      { version: 'live', notes: 'Сервер онлайн. Заходи по mc.petus.ru.' },
    ],
  },
];
